aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -825,7 +831,8 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
 
     // Scale q,k before matmul for stability see https://tinyurl.com/sudb9s96 for math
     const auto embed_size = SymFloat(query_.sym_size(-1));
-    const auto scaling_factor = embed_size.sqrt().sqrt();
+    const auto scaling_factor = scale.has_value() ? 1.0f / std::sqrt(scale.value())
```

Comment:
So, is it better to do `1.0f / std::sqrt(scale.value())` or to just do `scale.value()` and document it?

Reply:
Ah, but just below we're doing `query_ / scaling_factor`.

It's still strange that we're doing `query_ / (1.0f / std::sqrt(scale.value()))` instead of `query_ * std::sqrt(scale.value())`.

Reply:
you are right, I was having tunnel vision